# Qual Pais teve mais casos?
## Qual pais teve mais casos fatais?
# Qual ano teve mais casos?
## Qual ano teve mais casos fatais?
# Qual sexo teve mais casos?
## Qual sexo teve mais casos fatais?


In [743]:
import warnings
warnings.simplefilter("ignore")

In [771]:
import pandas as pd
import numpy as np
import re

In [772]:
df=pd.read_csv('attacks.csv',sep=',')

In [773]:
df

Case Number         Date    Year        Type    Country  \
0      2018.06.25  25-Jun-2018  2018.0     Boating        USA   
1      2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA   
2      2018.06.09  09-Jun-2018  2018.0     Invalid        USA   
3      2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA   
4      2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO   
...           ...          ...     ...         ...        ...   
25718         NaN          NaN     NaN         NaN        NaN   
25719         NaN          NaN     NaN         NaN        NaN   
25720         NaN          NaN     NaN         NaN        NaN   
25721         NaN          NaN     NaN         NaN        NaN   
25722          xx          NaN     NaN         NaN        NaN   

                  Area                        Location     Activity  \
0           California     Oceanside, San Diego County     Paddling   
1              Georgia  St. Simon Island, Glynn County     Standing   
2               Hawaii                    Habush, Oahu      Surfing   
3      New South Wales              Arrawarra Headland      Surfing   
4               Colima                        La Ticla  Free diving   
...                ...                             ...          ...   
25718              NaN                             NaN          NaN   
25719              NaN                             NaN          NaN   
25720              NaN                             NaN          NaN   
25721              NaN                             NaN          NaN   
25722              NaN                             NaN          NaN   

                  Name Sex   ...         Species   \
0          Julie Wolfe    F  ...      White shark   
1      Adyson�McNeely     F  ...              NaN   
2          John Denges    M  ...              NaN   
3                 male    M  ...        2 m shark   
4       Gustavo Ramos     M  ...  Tiger shark, 3m   
...                ...  ...  ...              ...   
25718              NaN  NaN  ...              NaN   
25719              NaN  NaN  ...              NaN   
25720              NaN  NaN  ...              NaN   
25721              NaN  NaN  ...              NaN   
25722              NaN  NaN  ...              NaN   

               Investigator or Source                       pdf  \
0                    R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1      K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2      K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3                      B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4                           A .Kipper      2018.06.04-Ramos.pdf   
...                               ...                       ...   
25718                             NaN                       NaN   
25719                             NaN                       NaN   
25720                             NaN                       NaN   
25721                             NaN                       NaN   
25722                             NaN                       NaN   

                                            href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...   
1      http://sharkattackfile.net/spreadsheets/pdf_di...   
2      http://sharkattackfile.net/spreadsheets/pdf_di...   
3      http://sharkattackfile.net/spreadsheets/pdf_di...   
4      http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                  ...   
25718                                                NaN   
25719                                                NaN   
25720                                                NaN   
25721                                                NaN   
25722                                                NaN   

                                                    href Case Number.1  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1      http://sharkattackfile.net/spreadsheets/pdf_di...    2

In [774]:
#Tratamento de valores nulos
df.loc[(df.isna().sum(axis=1))>=24]

Case Number Date  Year Type Country Area Location Activity Name Sex   \
8702          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
8703          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
8704          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
8705          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
8706          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
...           ...  ...   ...  ...     ...  ...      ...      ...  ...  ...   
25717         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
25718         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
25719         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
25720         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
25721         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   

       ... Species  Investigator or Source  pdf href formula href  \
8702   ...      NaN                    NaN  NaN          NaN  NaN   
8703   ...      NaN                    NaN  NaN          NaN  NaN   
8704   ...      NaN                    NaN  NaN          NaN  NaN   
8705   ...      NaN                    NaN  NaN          NaN  NaN   
8706   ...      NaN                    NaN  NaN          NaN  NaN   
...    ...      ...                    ...  ...          ...  ...   
25717  ...      NaN                    NaN  NaN          NaN  NaN   
25718  ...      NaN                    NaN  NaN          NaN  NaN   
25719  ...      NaN                    NaN  NaN          NaN  NaN   
25720  ...      NaN                    NaN  NaN          NaN  NaN   
25721  ...      NaN                    NaN  NaN          NaN  NaN   

      Case Number.1 Case Number.2 original order Unnamed: 22 Unnamed: 23  
8702            NaN           NaN            NaN         NaN         NaN  
8703            NaN           NaN            NaN         NaN         NaN  
8704            NaN           NaN            NaN         NaN         NaN  
8705            NaN           NaN            NaN         NaN         NaN  
8706            NaN           NaN            NaN         NaN         NaN  
...             ...           ...            ...         ...         ...  
25717           NaN           NaN            NaN         NaN         NaN  
25718           NaN           NaN            NaN         NaN         NaN  
25719           NaN           NaN            NaN         NaN         NaN  
25720           NaN           NaN            NaN         NaN         NaN  
25721           NaN           NaN            NaN         NaN         NaN  

[17020 rows x 24 columns]

In [775]:
df=df.dropna(how='all')
df

Case Number         Date    Year        Type    Country  \
0      2018.06.25  25-Jun-2018  2018.0     Boating        USA   
1      2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA   
2      2018.06.09  09-Jun-2018  2018.0     Invalid        USA   
3      2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA   
4      2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO   
...           ...          ...     ...         ...        ...   
8698            0          NaN     NaN         NaN        NaN   
8699            0          NaN     NaN         NaN        NaN   
8700            0          NaN     NaN         NaN        NaN   
8701            0          NaN     NaN         NaN        NaN   
25722          xx          NaN     NaN         NaN        NaN   

                  Area                        Location     Activity  \
0           California     Oceanside, San Diego County     Paddling   
1              Georgia  St. Simon Island, Glynn County     Standing   
2               Hawaii                    Habush, Oahu      Surfing   
3      New South Wales              Arrawarra Headland      Surfing   
4               Colima                        La Ticla  Free diving   
...                ...                             ...          ...   
8698               NaN                             NaN          NaN   
8699               NaN                             NaN          NaN   
8700               NaN                             NaN          NaN   
8701               NaN                             NaN          NaN   
25722              NaN                             NaN          NaN   

                  Name Sex   ...         Species   \
0          Julie Wolfe    F  ...      White shark   
1      Adyson�McNeely     F  ...              NaN   
2          John Denges    M  ...              NaN   
3                 male    M  ...        2 m shark   
4       Gustavo Ramos     M  ...  Tiger shark, 3m   
...                ...  ...  ...              ...   
8698               NaN  NaN  ...              NaN   
8699               NaN  NaN  ...              NaN   
8700               NaN  NaN  ...              NaN   
8701               NaN  NaN  ...              NaN   
25722              NaN  NaN  ...              NaN   

               Investigator or Source                       pdf  \
0                    R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1      K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2      K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3                      B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4                           A .Kipper      2018.06.04-Ramos.pdf   
...                               ...                       ...   
8698                              NaN                       NaN   
8699                              NaN                       NaN   
8700                              NaN                       NaN   
8701                              NaN                       NaN   
25722                             NaN                       NaN   

                                            href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...   
1      http://sharkattackfile.net/spreadsheets/pdf_di...   
2      http://sharkattackfile.net/spreadsheets/pdf_di...   
3      http://sharkattackfile.net/spreadsheets/pdf_di...   
4      http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                  ...   
8698                                                 NaN   
8699                                                 NaN   
8700                                                 NaN   
8701                                                 NaN   
25722                                                NaN   

                                                    href Case Number.1  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1      http://sharkattackfile.net/spreadsheets/pdf_di...    2

In [776]:
df=df.rename({'Sex ':'Sex'},axis=1)
df=df.rename({'Species ':'Species'},axis=1)

In [777]:
#Tratando 'Case Number' que tem valor '0' e 'xx'
df=df.loc[(df['Case Number'] != '0') & (df['Case Number'] != 'xx'),: ]
df

Case Number         Date    Year        Type             Country  \
0     2018.06.25  25-Jun-2018  2018.0     Boating                 USA   
1     2018.06.18  18-Jun-2018  2018.0  Unprovoked                 USA   
2     2018.06.09  09-Jun-2018  2018.0     Invalid                 USA   
3     2018.06.08  08-Jun-2018  2018.0  Unprovoked           AUSTRALIA   
4     2018.06.04  04-Jun-2018  2018.0    Provoked              MEXICO   
...          ...          ...     ...         ...                 ...   
6297     ND.0005  Before 1903     0.0  Unprovoked           AUSTRALIA   
6298     ND.0004  Before 1903     0.0  Unprovoked           AUSTRALIA   
6299     ND.0003    1900-1905     0.0  Unprovoked                 USA   
6300     ND.0002    1883-1889     0.0  Unprovoked              PANAMA   
6301     ND.0001    1845-1853     0.0  Unprovoked  CEYLON (SRI LANKA)   

                   Area                             Location      Activity  \
0            California          Oceanside, San Diego County      Paddling   
1               Georgia       St. Simon Island, Glynn County      Standing   
2                Hawaii                         Habush, Oahu       Surfing   
3       New South Wales                   Arrawarra Headland       Surfing   
4                Colima                             La Ticla   Free diving   
...                 ...                                  ...           ...   
6297  Western Australia                          Roebuck Bay        Diving   
6298  Western Australia                                  NaN  Pearl diving   
6299     North Carolina                       Ocracoke Inlet      Swimming   
6300                NaN                 Panama Bay 8�N, 79�W           NaN   
6301   Eastern Province  Below the English fort, Trincomalee      Swimming   

                       Name Sex  ...          Species  \
0               Julie Wolfe   F  ...      White shark   
1           Adyson�McNeely    F  ...              NaN   
2               John Denges   M  ...              NaN   
3                      male   M  ...        2 m shark   
4            Gustavo Ramos    M  ...  Tiger shark, 3m   
...                     ...  ..  ...              ...   
6297                   male   M  ...              NaN   
6298                  Ahmun   M  ...              NaN   
6299  Coast Guard personnel   M  ...              NaN   
6300        Jules Patterson   M  ...              NaN   
6301                   male   M  ...              NaN   

                     Investigator or Source                             pdf  \
0                          R. Collier, GSAF            2018.06.25-Wolfe.pdf   
1            K.McMurray, TrackingSharks.com          2018.06.18-McNeely.pdf   
2            K.McMurray, TrackingSharks.com           2018.06.09-Denges.pdf   
3                            B. Myatt, GSAF        2018.06.08-Arrawarra.pdf   
4                                 A .Kipper            2018.06.04-Ramos.pdf   
...                                     ...                             ...   
6297       H. Taunton; N. Bartlett,  p. 234          ND-0005-RoebuckBay.pdf   
6298  H. Taunton; N. Bartlett,  pp. 233-234               ND-0004-Ahmun.pdf   
6299   F. Schwartz, p.23; C. Creswell, GSAF  ND-0003-Ocracoke_1900-1905.pdf   
6300                    The Sun, 10/20/1938      ND-0002-JulesPatterson.pdf   
6301                             S.W. Baker              ND-0001-Ceylon.pdf   

                                           href formula  \
0     http://sharkattackfile.net/spreadsheets/pdf_di...   
1     http://sharkattackfile.net/spreadsheets/pdf_di...   
2     http://sharkattackfile.net/spreadsheets/pdf_di...   
3     http://sharkattackfile.net/spreadsheets/pdf_di...   
4     http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                 ...   
6297  http://sharkattackfile.net/spreadsheets/pdf_di...   
6298  http://sharkattackfile.net/spreadsheets/pdf_di...   
6299  http://sharkattackfi

In [832]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6302 entries, 0 to 6301
Data columns (total 24 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Case Number             6301 non-null   object 
 1   Date                    6302 non-null   object 
 2   Year                    6302 non-null   int32  
 3   Type                    6298 non-null   object 
 4   Country                 6252 non-null   object 
 5   Area                    5847 non-null   object 
 6   Location                5762 non-null   object 
 7   Activity                5758 non-null   object 
 8   Name                    6092 non-null   object 
 9   Sex                     6302 non-null   object 
 10  Age                     3471 non-null   object 
 11  Injury                  6274 non-null   object 
 12  Fatal (Y/N)             5763 non-null   object 
 13  Time                    2948 non-null   object 
 14  Species                 3464 non-null   

In [778]:
#Conferindo valors nulos no Unnamed 22 e 23 (como tem descrições relevantes precisa ser mantido)
df.loc[df['Unnamed: 22'].isnull()==False]

Case Number         Date    Year        Type Country    Area  \
1478  2006.05.27  27-May-2006  2006.0  Unprovoked     USA  Hawaii   

                Location Activity          Name Sex  ... Species  \
1478  North Shore, O'ahu  Surfing  Bret Desmond   M  ...     NaN   

     Investigator or Source                     pdf  \
1478             R. Collier  2006.05.27-Desmond.pdf   

                                           href formula  \
1478  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
1478  http://sharkattackfile.net/spreadsheets/pdf_di...    2006.05.27   

     Case Number.2 original order   Unnamed: 22 Unnamed: 23  
1478    2006.05.27         4825.0  stopped here         NaN  

[1 rows x 24 columns]

In [779]:
df.loc[df['Unnamed: 23'].isnull()==False]

Case Number                  Date    Year        Type  \
4415    1952.03.30           30-Mar-1952  1952.0  Unprovoked   
5840  1878.09.14.R  Reported 14-Sep-1878  1878.0    Provoked   

                   Country         Area                    Location  \
4415  NETHERLANDS ANTILLES      Curacao                         NaN   
5840                   USA  Connecticut  Branford, New Haven County   

                                             Activity              Name Sex  \
4415  Went to aid of child being menaced by the shark       A.J. Eggink   M   
5840                                          Fishing  Captain Pattison   M   

      ...                                            Species  \
4415  ...  Bull shark, 2.7 m [9'] was captured & dragged ...   
5840  ...                                                NaN   

                                 Investigator or Source  \
4415  J. Randall, p.352 in Sharks & Survival; H.D. B...   
5840                       St. Joseph Herald, 9/14/1878   

                            pdf  \
4415      1952.03.30-Eggink.pdf   
5840  1878.09.14.R-Pattison.pdf   

                                           href formula  \
4415  http://sharkattackfile.net/spreadsheets/pdf_di...   
5840  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
4415  http://sharkattackfile.net/spreadsheets/pdf_di...    1952.03.30   
5840  http://sharkattackfile.net/spreadsheets/pdf_di...  1878.09.14.R   

     Case Number.2 original order Unnamed: 22      Unnamed: 23  
4415    1952.03.30         1888.0         NaN           Teramo  
5840  1878.09.14.R          463.0         NaN  change filename  

[2 rows x 24 columns]

In [780]:
#Conferindo se tem duplicatas
df.duplicated().value_counts()

False    6302
dtype: int64

In [781]:
#tratamento de coluna 'Sex'
df['Sex'].value_counts()

M      5094
F       637
M         2
N         2
.         1
lli       1
Name: Sex, dtype: int64

In [782]:
df['Sex']=np.where(df['Sex']=='M ','M',df['Sex'])
df['Sex']=np.where(df['Name']=='Stephen Pettigew','M',df['Sex'])#1. William%Lesile Newton (NAN) #Stephen Petigew (Male)
df['Sex']=np.where(df['Sex']=='N',0,df['Sex'])
df['Sex']=np.where(df['Sex']=='lli','M',df['Sex'])
df['Sex']=np.where(df['Sex']=='.',0,df['Sex'])
df['Sex']=np.where(df['Sex'].isnull()==True,0,df['Sex'])

In [783]:
df

Case Number         Date    Year        Type             Country  \
0     2018.06.25  25-Jun-2018  2018.0     Boating                 USA   
1     2018.06.18  18-Jun-2018  2018.0  Unprovoked                 USA   
2     2018.06.09  09-Jun-2018  2018.0     Invalid                 USA   
3     2018.06.08  08-Jun-2018  2018.0  Unprovoked           AUSTRALIA   
4     2018.06.04  04-Jun-2018  2018.0    Provoked              MEXICO   
...          ...          ...     ...         ...                 ...   
6297     ND.0005  Before 1903     0.0  Unprovoked           AUSTRALIA   
6298     ND.0004  Before 1903     0.0  Unprovoked           AUSTRALIA   
6299     ND.0003    1900-1905     0.0  Unprovoked                 USA   
6300     ND.0002    1883-1889     0.0  Unprovoked              PANAMA   
6301     ND.0001    1845-1853     0.0  Unprovoked  CEYLON (SRI LANKA)   

                   Area                             Location      Activity  \
0            California          Oceanside, San Diego County      Paddling   
1               Georgia       St. Simon Island, Glynn County      Standing   
2                Hawaii                         Habush, Oahu       Surfing   
3       New South Wales                   Arrawarra Headland       Surfing   
4                Colima                             La Ticla   Free diving   
...                 ...                                  ...           ...   
6297  Western Australia                          Roebuck Bay        Diving   
6298  Western Australia                                  NaN  Pearl diving   
6299     North Carolina                       Ocracoke Inlet      Swimming   
6300                NaN                 Panama Bay 8�N, 79�W           NaN   
6301   Eastern Province  Below the English fort, Trincomalee      Swimming   

                       Name Sex  ...          Species  \
0               Julie Wolfe   F  ...      White shark   
1           Adyson�McNeely    F  ...              NaN   
2               John Denges   M  ...              NaN   
3                      male   M  ...        2 m shark   
4            Gustavo Ramos    M  ...  Tiger shark, 3m   
...                     ...  ..  ...              ...   
6297                   male   M  ...              NaN   
6298                  Ahmun   M  ...              NaN   
6299  Coast Guard personnel   M  ...              NaN   
6300        Jules Patterson   M  ...              NaN   
6301                   male   M  ...              NaN   

                     Investigator or Source                             pdf  \
0                          R. Collier, GSAF            2018.06.25-Wolfe.pdf   
1            K.McMurray, TrackingSharks.com          2018.06.18-McNeely.pdf   
2            K.McMurray, TrackingSharks.com           2018.06.09-Denges.pdf   
3                            B. Myatt, GSAF        2018.06.08-Arrawarra.pdf   
4                                 A .Kipper            2018.06.04-Ramos.pdf   
...                                     ...                             ...   
6297       H. Taunton; N. Bartlett,  p. 234          ND-0005-RoebuckBay.pdf   
6298  H. Taunton; N. Bartlett,  pp. 233-234               ND-0004-Ahmun.pdf   
6299   F. Schwartz, p.23; C. Creswell, GSAF  ND-0003-Ocracoke_1900-1905.pdf   
6300                    The Sun, 10/20/1938      ND-0002-JulesPatterson.pdf   
6301                             S.W. Baker              ND-0001-Ceylon.pdf   

                                           href formula  \
0     http://sharkattackfile.net/spreadsheets/pdf_di...   
1     http://sharkattackfile.net/spreadsheets/pdf_di...   
2     http://sharkattackfile.net/spreadsheets/pdf_di...   
3     http://sharkattackfile.net/spreadsheets/pdf_di...   
4     http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                 ...   
6297  http://sharkattackfile.net/spreadsheets/pdf_di...   
6298  http://sharkattackfile.net/spreadsheets/pdf_di...   
6299  http://sharkattackfi

In [784]:
#Tratamento de coluna 'Country'
country_list=list(df['Country'].unique())
country_list

['USA',
 'AUSTRALIA',
 'MEXICO',
 'BRAZIL',
 'ENGLAND',
 'SOUTH AFRICA',
 'THAILAND',
 'COSTA RICA',
 'MALDIVES',
 'BAHAMAS',
 'NEW CALEDONIA',
 'ECUADOR',
 'MALAYSIA',
 'LIBYA',
 nan,
 'CUBA',
 'MAURITIUS',
 'NEW ZEALAND',
 'SPAIN',
 'SAMOA',
 'SOLOMON ISLANDS',
 'JAPAN',
 'EGYPT',
 'ST HELENA, British overseas territory',
 'COMOROS',
 'REUNION',
 'FRENCH POLYNESIA',
 'UNITED KINGDOM',
 'UNITED ARAB EMIRATES',
 'PHILIPPINES',
 'INDONESIA',
 'CHINA',
 'COLUMBIA',
 'CAPE VERDE',
 'Fiji',
 'DOMINICAN REPUBLIC',
 'CAYMAN ISLANDS',
 'ARUBA',
 'MOZAMBIQUE',
 'FIJI',
 'PUERTO RICO',
 'ITALY',
 'ATLANTIC OCEAN',
 'GREECE',
 'ST. MARTIN',
 'FRANCE',
 'PAPUA NEW GUINEA',
 'TRINIDAD & TOBAGO',
 'KIRIBATI',
 'ISRAEL',
 'DIEGO GARCIA',
 'TAIWAN',
 'JAMAICA',
 'PALESTINIAN TERRITORIES',
 'GUAM',
 'SEYCHELLES',
 'BELIZE',
 'NIGERIA',
 'TONGA',
 'SCOTLAND',
 'CANADA',
 'CROATIA',
 'SAUDI ARABIA',
 'CHILE',
 'ANTIGUA',
 'KENYA',
 'RUSSIA',
 'TURKS & CAICOS',
 'UNITED ARAB EMIRATES (UAE)',
 'AZORES',
 

In [785]:
country_list=['USA',
 'AUSTRALIA',
 'MEXICO',
 'BRAZIL',
 'ENGLAND',
 'SOUTH AFRICA',
 'THAILAND',
 'COSTA RICA',
 'MALDIVES',
 'BAHAMAS',
 'NEW CALEDONIA',
 'ECUADOR',
 'MALAYSIA',
 'LIBYA',
 'CUBA',
 'MAURITIUS',
 'NEW ZEALAND',
 'SPAIN',
 'SAMOA',
 'SOLOMON ISLANDS',
 'JAPAN',
 'EGYPT',
 'ST HELENA, British overseas territory',
 'COMOROS',
 'REUNION',
 'FRENCH POLYNESIA',
 'UNITED KINGDOM',
 'UNITED ARAB EMIRATES',
 'PHILIPPINES',
 'INDONESIA',
 'CHINA',
 'COLUMBIA',
 'CAPE VERDE',
 'Fiji',
 'DOMINICAN REPUBLIC',
 'CAYMAN ISLANDS',
 'ARUBA',
 'MOZAMBIQUE',
 'FIJI',
 'PUERTO RICO',
 'ITALY',
 'ATLANTIC OCEAN',
 'GREECE',
 'ST. MARTIN',
 'FRANCE',
 'PAPUA NEW GUINEA',
 'TRINIDAD & TOBAGO',
 'KIRIBATI',
 'ISRAEL',
 'DIEGO GARCIA',
 'TAIWAN',
 'JAMAICA',
 'PALESTINIAN TERRITORIES',
 'GUAM',
 'SEYCHELLES',
 'BELIZE',
 'NIGERIA',
 'TONGA',
 'SCOTLAND',
 'CANADA',
 'CROATIA',
 'SAUDI ARABIA',
 'CHILE',
 'ANTIGUA',
 'KENYA',
 'RUSSIA',
 'TURKS & CAICOS',
 'AZORES',
 'SOUTH KOREA',
 'MALTA',
 'VIETNAM',
 'MADAGASCAR',
 'PANAMA',
 'SOMALIA',
 'NEVIS',
 'BRITISH VIRGIN ISLANDS',
 'NORWAY',
 'SENEGAL',
 'YEMEN',
 'GULF OF ADEN',
 'Sierra Leone',
 'GRAND CAYMAN',
 'Seychelles',
 'LIBERIA',
 'VANUATU',
 'HONDURAS',
 'VENEZUELA',
 'SRI LANKA',
 'URUGUAY',
 'INDIA',
 'MICRONESIA',
 'CARIBBEAN SEA',
 'OKINAWA',
 'TANZANIA',
 'MARSHALL ISLANDS',
 'EGYPT / ISRAEL',
 'NORTHERN ARABIAN SEA',
 'HONG KONG',
 'EL SALVADOR',
 'ANGOLA',
 'BERMUDA',
 'MONTENEGRO',
 'IRAN',
 'TUNISIA',
 'NAMIBIA',
 'NORTH ATLANTIC OCEAN',
 'PORTUGAL',
 'SOUTH CHINA SEA',
 'BANGLADESH',
 'PALAU',
 'WESTERN SAMOA',
 'PACIFIC OCEAN',
 'BRITISH ISLES',
 'GRENADA',
 'IRAQ',
 'TURKEY',
 'SINGAPORE',
 'NEW BRITAIN',
 'SUDAN',
 'JOHNSTON ISLAND',
 'SOUTH PACIFIC OCEAN',
 'NEW GUINEA',
 'RED SEA',
 'NORTH PACIFIC OCEAN',
 'FEDERATED STATES OF MICRONESIA',
 'MID ATLANTIC OCEAN',
 'ADMIRALTY ISLANDS',
 'BRITISH WEST INDIES',
 'SOUTH ATLANTIC OCEAN',
 'PERSIAN GULF',
 'RED SEA / INDIAN OCEAN',
 'NORTH SEA',
 'NICARAGUA',
 'AMERICAN SAMOA',
 'ANDAMAN / NICOBAR ISLANDAS',
 'GABON',
 'MAYOTTE',
 'THE BALKANS',
 'ARGENTINA',
 'MARTINIQUE',
 'INDIAN OCEAN',
 'GUATEMALA',
 'NETHERLANDS ANTILLES',
 'NORTHERN MARIANA ISLANDS',
 'IRAN / IRAQ',
 'JAVA',
 'SIERRA LEONE',
 'CENTRAL PACIFIC',
 'SOLOMON ISLANDS / VANUATU',
 'SOUTHWEST PACIFIC OCEAN',
 'BAY OF BENGAL',
 'MID-PACIFC OCEAN',
 'SLOVENIA',
 'CURACAO',
 'ICELAND',
 'ITALY / CROATIA',
 'BARBADOS',
 'MONACO',
 'GUYANA',
 'HAITI',
 'SAN DOMINGO',
 'IRELAND',
 'KUWAIT',
 'FALKLAND ISLANDS',
 'CRETE',
 'CYPRUS',
 'WEST INDIES',
 'BURMA',
 'LEBANON',
 'PARAGUAY',
 'BRITISH NEW GUINEA',
 'CEYLON',
 'OCEAN',
 'GEORGIA',
 'SYRIA',
 'TUVALU',
 'INDIAN OCEAN?',
 'GUINEA',
 'ANDAMAN ISLANDS',
 'EQUATORIAL GUINEA / CAMEROON',
 'COOK ISLANDS',
 'TOBAGO',
 'PERU',
 'AFRICA',
 'ALGERIA',
 'Coast of AFRICA',
 'TASMAN SEA',
 'GHANA',
 'GREENLAND',
 'MEDITERRANEAN SEA',
 'SWEDEN',
 'ROATAN',
 'Between PORTUGAL & INDIA',
 'DJIBOUTI',
 'BAHREIN',
 'ASIA']

In [786]:
country_list.sort()
country_list

['ADMIRALTY ISLANDS',
 'AFRICA',
 'ALGERIA',
 'AMERICAN SAMOA',
 'ANDAMAN / NICOBAR ISLANDAS',
 'ANDAMAN ISLANDS',
 'ANGOLA',
 'ANTIGUA',
 'ARGENTINA',
 'ARUBA',
 'ASIA',
 'ATLANTIC OCEAN',
 'AUSTRALIA',
 'AZORES',
 'BAHAMAS',
 'BAHREIN',
 'BANGLADESH',
 'BARBADOS',
 'BAY OF BENGAL',
 'BELIZE',
 'BERMUDA',
 'BRAZIL',
 'BRITISH ISLES',
 'BRITISH NEW GUINEA',
 'BRITISH VIRGIN ISLANDS',
 'BRITISH WEST INDIES',
 'BURMA',
 'Between PORTUGAL & INDIA',
 'CANADA',
 'CAPE VERDE',
 'CARIBBEAN SEA',
 'CAYMAN ISLANDS',
 'CENTRAL PACIFIC',
 'CEYLON',
 'CHILE',
 'CHINA',
 'COLUMBIA',
 'COMOROS',
 'COOK ISLANDS',
 'COSTA RICA',
 'CRETE',
 'CROATIA',
 'CUBA',
 'CURACAO',
 'CYPRUS',
 'Coast of AFRICA',
 'DIEGO GARCIA',
 'DJIBOUTI',
 'DOMINICAN REPUBLIC',
 'ECUADOR',
 'EGYPT',
 'EGYPT / ISRAEL',
 'EL SALVADOR',
 'ENGLAND',
 'EQUATORIAL GUINEA / CAMEROON',
 'FALKLAND ISLANDS',
 'FEDERATED STATES OF MICRONESIA',
 'FIJI',
 'FRANCE',
 'FRENCH POLYNESIA',
 'Fiji',
 'GABON',
 'GEORGIA',
 'GHANA',
 'GRAND CAYM

In [787]:
df['Country']=np.where(df['Country']==' PHILIPPINES','PHILIPPINES',df['Country'])
df['Country']=np.where(df['Country']==' TONGA','TONGA',df['Country'])
df['Country']=np.where(df['Country']=='ASIA?','ASIA',df['Country'])
df['Country']=np.where(df['Country']=='CEYLON (SRI LANKA)','CEYLON',df['Country'])
df['Country']=np.where(df['Country']=='EGYPT ','EGYPT',df['Country'])
df['Country']=np.where(df['Country']=='MALDIVE ISLANDS','MALDIVES',df['Country'])
df['Country']=np.where(df['Country']=='MEXICO ','MEXICO',df['Country'])
df['Country']=np.where(df['Country']=='NICARAGUA ','NICARAGUA',df['Country'])
df['Country']=np.where(df['Country']=='NORTH ATLANTIC OCEAN ','NORTH ATLANTIC OCEAN',df['Country'])
df['Country']=np.where(df['Country']=='PACIFIC OCEAN ','PACIFIC OCEAN',df['Country'])
df['Country']=np.where(df['Country']=='RED SEA?','RED SEA',df['Country'])
df['Country']=np.where(df['Country']=='REUNION ISLAND','REUNION',df['Country'])
df['Country']=np.where(df['Country']=='KOREA','SOUTH KOREA',df['Country'])
df['Country']=np.where(df['Country']=='ST. MAARTIN','ST. MARTIN',df['Country'])
df['Country']=np.where(df['Country']=='SUDAN?','SUDAN',df['Country'])
df['Country']=np.where(df['Country']=='UNITED ARAB EMIRATES (UAE)','UNITED ARAB EMIRATES',df['Country'])
df['Country']=np.where(df['Country']=='YEMEN ','YEMEN',df['Country'])

In [788]:
df['Year'].isnull().value_counts()

False    6300
True        2
Name: Year, dtype: int64

In [789]:
#transformando ano conforme ao Date
df['Year']=np.where((df['Year'].isnull()==True) & (df['Type']=='Invalid'),2017,df['Year']) 
df['Year']=np.where((df['Year'].isnull()==True) & (df['Type']=='Unprovoked'),1836,df['Year'])


In [807]:
df.Year=df.Year.astype(int)

In [808]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6302 entries, 0 to 6301
Data columns (total 24 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Case Number             6301 non-null   object 
 1   Date                    6302 non-null   object 
 2   Year                    6302 non-null   int32  
 3   Type                    6298 non-null   object 
 4   Country                 6252 non-null   object 
 5   Area                    5847 non-null   object 
 6   Location                5762 non-null   object 
 7   Activity                5758 non-null   object 
 8   Name                    6092 non-null   object 
 9   Sex                     6302 non-null   object 
 10  Age                     3471 non-null   object 
 11  Injury                  6274 non-null   object 
 12  Fatal (Y/N)             5763 non-null   object 
 13  Time                    2948 non-null   object 
 14  Species                 3464 non-null   

In [809]:
df['Fatal (Y/N)'].value_counts()

N    4303
Y    1389
0      71
Name: Fatal (Y/N), dtype: int64

In [810]:
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']==' N','N',df['Fatal (Y/N)'])
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='y','Y',df['Fatal (Y/N)'])
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='N ','N',df['Fatal (Y/N)'])
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='M','N',df['Fatal (Y/N)']) #Erro de digitação
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='2017','N',df['Fatal (Y/N)']) #Pelo contexto não parece ser fatal
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='UNKNOWN',0,df['Fatal (Y/N)']) #transformar unknown para o

In [811]:
df_filtered=df[['Year','Country','Activity','Sex','Injury','Fatal (Y/N)','Species']]

 1. Qual pais teve mais casos?

In [812]:
df_filtered

Year    Country      Activity Sex  \
0     2018        USA      Paddling   F   
1     2018        USA      Standing   F   
2     2018        USA       Surfing   M   
3     2018  AUSTRALIA       Surfing   M   
4     2018     MEXICO   Free diving   M   
...    ...        ...           ...  ..   
6297     0  AUSTRALIA        Diving   M   
6298     0  AUSTRALIA  Pearl diving   M   
6299     0        USA      Swimming   M   
6300     0     PANAMA           NaN   M   
6301     0     CEYLON      Swimming   M   

                                                 Injury Fatal (Y/N)  \
0     No injury to occupant, outrigger canoe and pad...           N   
1                            Minor injury to left thigh           N   
2          Injury to left lower leg from surfboard skeg           N   
3                             Minor injury to lower leg           N   
4     Lacerations to leg & hand shark PROVOKED INCIDENT           N   
...                                                 ...         ...   
6297                                              FATAL           Y   
6298                                              FATAL           Y   
6299                                              FATAL           Y   
6300                                              FATAL           Y   
6301  FATAL. "Shark bit him in half, carrying away t...           Y   

              Species  
0         White shark  
1                 NaN  
2                 NaN  
3           2 m shark  
4     Tiger shark, 3m  
...               ...  
6297              NaN  
6298              NaN  
6299              NaN  
6300              NaN  
6301              NaN  

[6302 rows x 7 columns]

In [813]:
df_filtered['Country'].value_counts()

USA                 2229
AUSTRALIA           1338
SOUTH AFRICA         579
PAPUA NEW GUINEA     134
NEW ZEALAND          128
                    ... 
TUVALU                 1
KUWAIT                 1
ANGOLA                 1
WESTERN SAMOA          1
ANDAMAN ISLANDS        1
Name: Country, Length: 196, dtype: int64

Resposta : USA teve mais casos (2229 casos)

 1.1 Qual pais teve mais casos fatais

In [814]:
df_fatal=df_filtered.loc[df_filtered['Fatal (Y/N)']=='Y',:]
df_fatal[['Year','Fatal (Y/N)','Country']].groupby(by=['Fatal (Y/N)','Country']).count().sort_values(by='Year', ascending=False)

Year
Fatal (Y/N) Country               
Y           AUSTRALIA          283
            USA                186
            SOUTH AFRICA       106
            PAPUA NEW GUINEA    56
            MEXICO              43
...                            ...
            CYPRUS               1
            SAMOA                1
            SAN DOMINGO          1
            SAUDI ARABIA         1
            AFRICA               1

[142 rows x 1 columns]

In [815]:
total_fatal=len(df_fatal.loc[df_fatal['Fatal (Y/N)']=='Y','Fatal (Y/N)'])
total_fatal

1389

In [816]:
australia_fatal=len(df_fatal.loc[(df_fatal['Fatal (Y/N)']=='Y') & (df_fatal['Country']=='AUSTRALIA'),'Fatal (Y/N)'])
australia_fatal

283

In [864]:
australia_porc=australia_fatal/total_fatal*100
australia_porc

20.374370050395967

Resposta : Australia teve mais casos fatais (283 casos(20.4% dos casos))

2. Em que ano teve mais casos?

In [846]:
df_filtered['Year'].value_counts()

2015    143
2017    137
2016    130
2011    128
2014    127
       ... 
1786      1
1758      1
1742      1
1738      1
1554      1
Name: Year, Length: 249, dtype: int64

Resposta : Em 2015 teve mais casos (143 casos)

2.1 Qual pais que teve mais casos no ano que referente?

In [860]:
df_filtered.loc[df_filtered['Year']==2015,'Country'].value_counts()

USA                 73
AUSTRALIA           31
SOUTH AFRICA        10
REUNION              4
FRENCH POLYNESIA     4
MOZAMBIQUE           3
BAHAMAS              3
SPAIN                2
BRAZIL               2
ECUADOR              1
PUERTO RICO          1
NEW CALEDONIA        1
ITALY                1
COSTA RICA           1
MEXICO               1
ARUBA                1
EGYPT                1
THAILAND             1
FIJI                 1
ATLANTIC OCEAN       1
Name: Country, dtype: int64

Resposta : USA (73 casos)

2.2 Em que ano teve mais casos fatais?

In [853]:
df_fatal['Year'].value_counts()[1:]

1959    24
1956    21
1958    18
2000    18
1944    17
        ..
1786     1
1785     1
1784     1
1783     1
2018     1
Name: Year, Length: 219, dtype: int64

Resposta :Em 1959 teve mais casos fatais

2.3 Em que pais teve mais casos fatais no ano referente?

In [863]:
df_fatal.loc[df_fatal['Year']==1959,:'Country'].value_counts()

Year  Country         
1959  MEXICO              3
      PHILIPPINES         3
      USA                 3
      AUSTRALIA           2
      INDIA               2
      JAPAN               2
      MOZAMBIQUE          2
      PAPUA NEW GUINEA    2
      CARIBBEAN SEA       1
      FIJI                1
      MALDIVES            1
      SOLOMON ISLANDS     1
      SOUTH KOREA         1
dtype: int64

Resposta : Mexico (3 casos), Philippines (3 casos), USA (3 casos) 

3. Qual é a relação entre sexo do tubarão e fatalidade

In [899]:
df_corr=df_filtered[['Sex','Fatal (Y/N)']]
df_corr

Sex Fatal (Y/N)
0      F           N
1      F           N
2      M           N
3      M           N
4      M           N
...   ..         ...
6297   M           Y
6298   M           Y
6299   M           Y
6300   M           Y
6301   M           Y

[6302 rows x 2 columns]

In [900]:
df_onehotencoding=df_corr
df_onehotencoding['Male']=np.where(df_filtered['Sex']=='M',1,0)
df_onehotencoding['Female']=np.where(df_filtered['Sex']=='F',0,1)

df_onehotencoding['Fatal']=np.where(df_filtered['Fatal (Y/N)']=='Y',1,0)
df_onehotencoding.drop('Sex',axis=1, inplace=True)

In [906]:
df_onehotencoding

Male  Female  Fatal
0        0       0      0
1        0       0      0
2        1       1      0
3        1       1      0
4        1       1      0
...    ...     ...    ...
6297     1       1      1
6298     1       1      1
6299     1       1      1
6300     1       1      1
6301     1       1      1

[6302 rows x 3 columns]

In [907]:
df_onehotencoding.corr()[['Fatal']]

Fatal
Male    0.049046
Female  0.042415
Fatal   1.000000